pandas/io/formats/html.py

```diff
@@ -57,8 +57,11 @@ def __init__(
         self.bold_rows = self.fmt.bold_rows
         self.escape = self.fmt.escape
         self.show_dimensions = self.fmt.show_dimensions
-        if border is None:
+        if border is None or (border and isinstance(border, bool)):
             border = cast(int, get_option("display.html.border"))
+        elif not border and isinstance(border, bool):
```

Comment:
why is this instance check needed?

Reply:
Not my first choice, but it's to pass this test case: https://github.com/pandas-dev/pandas/blob/03ecf60e45ebcce5f1a548423a9b71ef737b5d19/pandas/tests/io/formats/test_to_html.py#L345

Setting `border=0` has to result in `border="0"'` being present in the output HTML. The instance check is to ensure that only a boolean can be used to hide the border.

Reply:
why is `border=0` not de-facto the same as `border=False`?

Reply:
In my opinion, it should be. That's how I originally wrote the fix. My only reason for taking this approach is to conform to the test case, which explicitly tests for a border attribute to be present upon `border=0`.

Reply:
i think ok to treat these cases as the same and can simplify the code a bit

Reply:
Sounds good to me - I will revert to my original fix, and update that test case instead